File: Java/_119_Pascal's_Triangle_II/Solution.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> curRow = new ArrayList<Integer>();
        
        if (rowIndex < 0)
            return curRow;
        curRow.add(1);
        
        for (int row = 0; row < rowIndex; row++) 
        {
            // push in 1 at the end of each row
            curRow.add(1);
            // construct each row from back to front
            for (int i = row; i >= 1; i--) 
                curRow.set(i, curRow.get(i) + curRow.get(i - 1));
        }
        return curRow;
    }
}

File: Python/_119_Pascal's_Triangle_II/solution.py
class Solution:
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        curRow = []
        if rowIndex < 0:
            return curRow

        curRow.append(1)
        for row in range(0, rowIndex):
            curRow.append(1)
            for i in range(1, row + 1)[::-1]:
                curRow[i] = curRow[i] + curRow[i - 1]
        return curRow
